components/validation/validation-custom.js

```diff
@@ -0,0 +1,18 @@
+
+import { LitElement } from 'lit-element/lit-element.js';
+import { ValidationCustomMixin } from './validation-custom-mixin.js';
+
+class ValidationCustom extends ValidationCustomMixin(LitElement) {
+
+	async validate() {
+		const validation = new Promise(resolve => {
+			const details = { detail: { target: this.target, resolve } };
+			const event = new CustomEvent('d2l-validation-custom-validate', details);
+			return this.dispatchEvent(event);
```

Comment:
What would it look like for someone to handle this event in practice? Something like...

```html
<d2l-validation-custom @d2l-validation-custom-validate="${this._validate}">
```

And then:
```javascript
_validate(e) {
    const val = e.detail.target.value;
    const isValid = val < 10 && val > 4;
    e.detail.resolve(isValid);
}
```

It'd be amazing if they could just return `true`, `false` or a promise from their event handler, but I can't think of a way to make that work. Plus I guess the way it's set up now, in theory there could be _many_ event handlers and only one of them needs to handle it (and could use `preventDefault` to stop the others), so... cool!

Also, events have targets already (`e.target` in this case, which will be the `d2l-validation-custom` element), so it's a little confusing for it to have `e.target` and `e.detail.target`. Trying to think of a better name... something like `forElement` maybe?

Reply:
Yep, thats exactly how it would be implemented. I did try to come up with options to just allow the value to be returned rather than having to pass in a `resolve` callback but as soon as the validate handler becomes `async` `e.preventDefault` doesn't work and return values can't be received anymore.

I'm happy renaming `target` to something else to avoid confusion.